### PR TITLE
[TASK] Make records hidable and timeable

### PR DESCRIPTION
**Pull request by lukaszuznanski**
Resolves: #553

### CODE REVIEW

**Comment by lukaszuznanski** [comment]
Different TCA needed for v11 and for v12

**Review by oliverklee — CHANGES_REQUESTED** [review]
I've manually tested this with my 11LTS testing installation:

- Hiding works if I use the "hide" icon in the list.
- When I edit a tea record in the DB, the fields hidden, starttime and endtime are not visible yet in the TCEforms.

So we'll also need to add those three fields to the `types` configuration.

**Review by oliverklee — CHANGES_REQUESTED** [review]
With this change, I get this exception when I try to edit a tea record in the BE:

> Item 0 of field hidden of TCA table tx_tea_domain_model_product_tea has no label

**Comment by lukaszuznanski** [comment]
@oliverklee ok, so it's different for v12 and for v11, I have to adjust configuration. It works fine on v12.

File: Configuration/TCA/tx_tea_domain_model_product_tea.php [hunk 1]
```diff
@@ -10,11 +10,53 @@
...
+            'config' => [
+                'type' => 'check',
+                'renderType' => 'checkboxToggle',
+                'items' => [
+                    [
+                        'label' => '',
+                        'invertStateDisplay' => true,
+                    ],
+                ],
+            ],
```
Comment (oliverklee) [inline Configuration/TCA/tx_tea_domain_model_product_tea.php:35]: This now causes the toggle checkbox to work in inverse. With this setting, the record is visible for me:

grafik

Comment (oliverklee) [reply Configuration/TCA/tx_tea_domain_model_product_tea.php:35]: This is what the TCA for core records looks like in V11:

```
        'hidden' => [
            'exclude' => true,
            'label' => 'LLL:EXT:core/Resources/Private/Language/locallang_general.xlf:LGL.enabled',
            'config' => [
                'type' => 'check',
                'renderType' => 'checkboxToggle',
                'items' => [
                    [
                        0 => '',
                        'invertStateDisplay' => true,
                    ],
                ],
            ],
        ],
```

Comment (oliverklee) [reply Configuration/TCA/tx_tea_domain_model_product_tea.php:35]: Strange, this is the same as our code. I don't understand this.

**Review by oliverklee — CHANGES_REQUESTED** [review]

**Comment by oliverklee** [comment]
Testing in 11LTS, the "hide" toggle still is the wrong way around.

File: Configuration/TCA/tx_tea_domain_model_product_tea.php [hunk 2]
```diff
@@ -10,11 +10,53 @@
...
+        'hidden' => [
+            'exclude' => true,
+            'label' => 'LLL:EXT:core/Resources/Private/Language/locallang_general.xlf:LGL.enabled',
+            'config' => [
+                'type' => 'check',
+                'renderType' => 'checkboxToggle',
+                'items' => [
+                    [
+                        'label' => '',
+                        'invertStateDisplay' => true,
```
Comment (oliverklee) [inline Configuration/TCA/tx_tea_domain_model_product_tea.php:32]: Ah, we also need this added after line 136:

```
'invertStateDisplay' => true,
```

**Review by oliverklee — CHANGES_REQUESTED** [review]

**Comment by oliverklee** [comment]
This still needs the V11 fix, and also a reabse.

**Comment by oliverklee** [comment]
@lukaszuznanski Do you have time to get this PR finished, or would you like me to take over?

**Comment by lukaszuznanski** [comment]
@oliverklee if you have time to finish it, then go for it

**Comment by oliverklee** [comment]
This looks really hard to rebase (also because we've merged `main` into this branch instead of rebasing it). I'm willing to do the V11 fixed, but don't have the time to first untangle the rebase mess right now.

**Comment by lukaszuznanski** [comment]
I think it might be easier and faster to open new PR with those changes, will do.

**Comment by oliverklee** [comment]
> I think it might be easier and faster to open new PR with those changes, will do.

Thanks!

**Comment by lukaszuznanski** [comment]
Closed as there is new PR #989

**Pull request closed without merging**